openpype/hosts/max/plugins/publish/validate_attributes.py

```diff
@@ -0,0 +1,131 @@
+
+from openpype.pipeline.publish import (
+    OptionalPyblishPluginMixin,
+    PublishValidationError,
+    RepairContextAction
+)
+
+
+def has_property(object_name, property_name):
+    """Return whether an object has a property with given name"""
+    return rt.Execute(f'isProperty {object_name} "{property_name}"')
+
+
+def is_matching_value(object_name, property_name, value):
+    """Return whether an existing property matches value `value"""
+    property_value = rt.Execute(f"{object_name}.{property_name}")
+
+    # Wrap property value if value is a string valued attributes
+    # starting with a `#`
+    if (
+        isinstance(value, str) and
+        value.startswith("#") and
+        not value.endswith(")")
+    ):
+        # prefix value with `#`
+        # not applicable for #() array value type
+        # and only applicable for enum i.e. #bob, #sally
+        property_value = f"#{property_value}"
+
+    return property_value == value
+
+
+class ValidateAttributes(OptionalPyblishPluginMixin,
+                         ContextPlugin):
+    """Validates attributes in the project setting are consistent
+    with the nodes from MaxWrapper Class in 3ds max.
+    E.g. "renderers.current.separateAovFiles",
+         "renderers.production.PrimaryGIEngine"
+    Admin(s) need to put json below and enable this validator for a check:
+    {
+       "renderers.current":{
+            "separateAovFiles" : True
+        }
+        "renderers.production":{
+            "PrimaryGIEngine": "#RS_GIENGINE_BRUTE_FORCE"
+        }
+        ....
+    }
+
+    """
+
+    order = ValidatorOrder
+    hosts = ["max"]
+    label = "Attributes"
+    actions = [RepairContextAction]
+    optional = True
+
+    @classmethod
+    def get_invalid(cls, context):
+        attributes = (
+            context.data["project_settings"]["max"]["publish"]
+                        ["ValidateAttributes"]["attributes"]
+        )
+        if not attributes:
+            return
+        invalid = []
+        for object_name, required_properties in attributes.items():
+            if not rt.Execute(f"isValidValue {object_name}"):
+                # Skip checking if the node does not
+                # exist in MaxWrapper Class
+                cls.log.debug(f"Unable to find '{object_name}'."
+                              " Skipping validation of attributes.")
+                continue
+
+            for property_name, value in required_properties.items():
+                if not has_property(object_name, property_name):
+                    cls.log.error(
+                        "Non-existing property: "
+                        f"{object_name}.{property_name}")
+                    invalid.append((object_name, property_name))
+
+                if not is_matching_value(object_name, property_name, value):
+                    cls.log.error(
+                        f"Invalid value for: {object_name}.{property_name}"
+                        f" Should be: {value}")
```

Comment:
With the removed dot it feels weird that `Should` is still uppercased?
```suggestion
                        f"Invalid value for: {object_name}.{property_name}"
                        f" should be: {value}")
```